test/test_images/probe_helper/probe/handler.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"context"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+/*
+
+The following steps should be taken when implementing a new type of probe.
+
+1. Define a static probe handler which implements the Handler interface.
+2. Declare a singleton Handler object of this kind in the probe helper initializerHandlers method.
+3. Add the mapping between forward probe type and the object from step 2 to the forwardProbeHandlers map in initializerHandlers.
+4. Add the mapping between receiver probe type and the object from step 2 to the receiveProbeHandlers map in initializerHandlers.
+
+*/
+
+// Handler is the interface which static probe objects should implement.
+type Handler interface {
+	// Forward is the handler function which is called for probe requests which come
+	// from PROBE_PORT.
+	Forward(context.Context, cloudevents.Event) error
+	// Receive is the handler function which is called for probe requests which come
+	// from RECEIVER_PORT.
+	Receive(context.Context, cloudevents.Event) error
+}
+
+func channelID(namespace, eventID string) string {
```

Comment:
nit: the first string is the targetPath. Why don't define the helper function like:
func channelID(event Event) string?

Reply:
The problem is that we want the channelID to match between forwarded and received events, but they come in different forms. In particular, the event.ID() of the forwarded event is not necessarily the event.ID() of the received event, the event must be parsed in a way specific to that event type which is specified in the Handler Receive method.

Reply:
We can avoid ever dealing with the channelID by just indexing into each Handler's personal receivedEvents map, at the index of the targetpath/receiverpath. In practice, this would look like, for each probe, indexing at key "/<namespace>". So as long as we can guarantee that there is only one probe of a certain kind running in a given namespace at a given time, this would still be safe. This isn't a problem with the dynamic namespace probes, but could easily be overlooked in the future for the static namespace probes, if the prober design changes.

Reply:
Please rename "namespace" to something like "prefix"? namespace is confusing since what's passed in is not a namespace.